test/e2e/main/main_test.go

```diff
@@ -817,3 +818,73 @@ func throwHTTPError(resp *http.Response) {
 	errorBytes, _ := ioutil.ReadAll(resp.Body)
 	panic(fmt.Errorf("unexpected HTTP status code (%d): %s", resp.StatusCode, string(errorBytes)))
 }
+
+func TestCacheCR(t *testing.T) {
+	spec := tutils.DefaultSpec(testKube)
+	name := strcase.ToKebab(t.Name())
+	spec.Name = name
+	testKube.CreateInfinispan(spec, tutils.Namespace)
+	testKube.WaitForInfinispanPods(1, tutils.SinglePodTimeout, spec.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(spec.Name, spec.Namespace, ispnv1.ConditionWellFormed)
+	cacheNameTemplateName := "testtemplatename"
+	cacheNameTemplateXML := "testtemplatexml"
+
+	ispn := ispnv1.Infinispan{}
+	testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Namespace: spec.Namespace, Name: spec.Name}, &ispn)
+
+	testCacheWithCR(cacheNameTemplateName, &ispn, getCacheCRwithTemplateName(cacheNameTemplateName, spec.Namespace, name))
+
+	testCacheWithCR(cacheNameTemplateXML, &ispn, getCacheCRwithTemplateXML(cacheNameTemplateXML, spec.Namespace, name))
+}
+
+func testCacheWithCR(cacheName string, ispn *ispnv1.Infinispan, cache *v2alpha1.Cache) {
+	key := "testkey"
+	value := "test-operator"
+	testKube.Create(cache)
+	//testKube.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Namespace: spec.Namespace, Name: spec.Name}, &ispn)
```

Comment:
Can be removed.

Reply:
@ryanemerson can you please review the changes?